### PR TITLE
Add a toggle option for typing indicators; use short ttl for them

### PR DESCRIPTION
**Pull request by msgmaxim**
- enables typing indicators by default (and adds an option to disable them)
- changes ttl for typing indicators to 1 min

### CODE REVIEW

File: settings.html [hunk 1]
```diff
@@ -120,6 +120,10 @@ <h3>{{ permissions }}</h3>
         <input type='checkbox' name='read-receipt-setting' id='read-receipt-setting' />
         <label for='read-receipt-setting'>{{ readReceiptSettingDescription }}</label>
       </div>
+      <div class='typing-indicators-setting'>
+        <input type='checkbox' name='typing-indicators-setting' id='typing-indicators-setting' /> 
+        <label for='typing-indicators-setting'>{{ typingIndicatorsSettingDescription }}</label>
+      </dev>
```
Comment (sachaaaaa) [inline settings.html]:
```suggestion
      </div>
```

File: js/models/conversations.js [hunk 2]
```diff
@@ -273,7 +273,8 @@
       const recipientId = this.isPrivate() ? this.id : null;
       const groupNumbers = this.getRecipients();
 
-      const sendOptions = this.getSendOptions();
+      let sendOptions = this.getSendOptions();
+      sendOptions.messageType = "typing";
```
Comment (sachaaaaa) [inline js/models/conversations.js]: I think our linter prefers single-quotes.
Have you tried running `yarn lint` before pushing?

Comment (msgmaxim) [reply js/models/conversations.js]: Yeah, I forgot that, and then noticed that travis complained :(

**Comment by sachaaaaa** [comment]
I wonder if typing indicators would only make sense via p2p? If we remove the "realtimeness" of the indicators, there is not much value left to it...

**Comment by msgmaxim** [comment]
They make sense when you have a real-time conversation with someone, but for whatever reason p2p is not available. 1 min ttl will ensure that we don't waste storage and that we don't get outdated indicators.

File: settings.html (hunk 1 above)
Comment (msgmaxim) [reply settings.html]: updated

File: js/models/conversations.js (hunk 2 above)
Comment (msgmaxim) [reply js/models/conversations.js]: updated

File: js/models/conversations.js [hunk 3]
```diff
@@ -273,7 +273,8 @@
       const recipientId = this.isPrivate() ? this.id : null;
       const groupNumbers = this.getRecipients();
 
-      const sendOptions = this.getSendOptions();
+      let sendOptions = this.getSendOptions();
```
Comment (sachaaaaa) [inline js/models/conversations.js]: javascript... 🤦‍♂ 
You can use `const` even if you mutate the object attributes, as long as you don't overwrite the object itself.

Comment (msgmaxim) [reply js/models/conversations.js]: updated

**Pull request merged**